docs/tutorials/first_app/blinky_cli.md

```diff
@@ -1,128 +1,165 @@
+The `--flash` argument will automatically flash the compiled program onto your board if it is connected to your computer. You can see which boards are connected with `mbed detect`. 
 
-Your PC may take a few minutes to compile your code. At the end you should get the following result:
+After the program has been flashed to the board, press the board's "reset" button and you should now see the LED blinking.
 
-```
-[snip]
-+----------------------------+-------+-------+------+
-| Module                     | .text | .data | .bss |
-+----------------------------+-------+-------+------+
-| Misc                       | 13939 |    24 | 1372 |
-| core/hal                   | 16993 |    96 |  296 |
-| core/rtos                  |  7384 |    92 | 4204 |
-| features/FEATURE_IPV4      |    80 |     0 |  176 |
-| frameworks/greentea-client |  1830 |    60 |   44 |
-| frameworks/utest           |  2392 |   512 |  292 |
-| Subtotals                  | 42618 |   784 | 6384 |
-+----------------------------+-------+-------+------+
-Allocated Heap: unknown
-Allocated Stack: unknown
-Total Static RAM memory (data + bss): 7168 bytes
-Total RAM memory (data + bss + heap + stack): 7168 bytes
-Total Flash memory (text + data + misc): 43402 bytes
-Image: .\.build\K64F\ARM\mbed-os-example-blinky.bin             
-```
+**Note** : you can get the name of the board plugged into your computer by running `mbed detect` and you can get a full list of supported toolchains / targets by running the `mbed compile --supported`
+
+
+### Debug
+
+#### Desktop IDE
+
+To debug using a desktop IDE such as Keil uVision, IAR, or Eclipse you can use the `mbed export` command to generate project files. For example, for a K64F and Keil uVision:
+
+```console
+$ mbed export --ide uvision --target K64F
+```  
+**NOTE** for a full list of exporters supported run the `mbed export --supported` command. 
+
+
+#### Printf
+
+Another way to do basic debugging is to use the `printf` command in your code, then read the output using a serial terminal such as [PuTTY](http://www.putty.org/) or [CoolTerm](http://freeware.the-meiers.org/). For example, add `printf("Hello World!\n\r");` to the top of your main function, then recompile the program and flash it to your device.
 
-The program file, `mbed-os-example-blinky.bin`, is under your `mbed-os-example-blinky\build\K64F\ARM\` folder.
+Invoke `mbed detect` from your command line to determine which communication port your board is connected to (i.e. `COM18`, `/dev/ttyACM0`, etc.). Unless otherwise specified, `printf` defaults to a baud rate/speed of `9600` on mbed OS.
 
-### Programming your board
+### Further Reading
 
-Arm Mbed Enabled boards are programmable by drag and drop over a USB connection:
+- Documentation
+  - [Mbed OS API's](https://os.mbed.com/docs/v5.6/reference/apis.html) - List of all API's available in Mbed OS
+  - [Peripheral Drivers](https://os.mbed.com/docs/v5.6/reference/drivers.html) - IO Driver API's (I2C, SPI, UART, ... etc)
```

Comment:
Query: Why do we link to the APIs and then link to a subsection of the APIs? They're included in the link above.

Reply:
This is based on interactions with users. majority of users still are interested in the 2.0 API's, now called peripheral drivers, so we link directly to them. Never hurts to have multiple paths

Reply:
agree to disagree and let it slide.